### PR TITLE
feat: get rid of createScopedRequired

### LINKED ISSUE #264: Remove createScopedRequired and createScope methods
Part of the feedback in #55 was to re-rationalize the createScopedRequired and createScope methods.  I wrote #255 to try addressing this, but it was way too early for a semver major change after going 1.0.

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Resolves #264

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
I am very confused and do not know why its showing all these commits.  "get rid of createScopedRequired" should be the only one.

File: test/test.googleauth.ts [hunk 1]
```diff
@@ -1229,6 +1230,13 @@ it('should accept credentials to get a client', async () => {
   assert.equal(client.email, 'hello@youarecool.com');
 });
 
+it('should allow passing scopes to get a client', async () => {
+  const scopes = ['http://examples.com/is/a/scope'];
```
Comment (jinwoo) [inline test/test.googleauth.ts]: `scopes` can be an array or a string, right? Maybe it's worthwhile to have a test for the latter case too?

File: README.md [hunk 2]
```diff
@@ -52,12 +52,14 @@ Rather than manually creating an OAuth2 client, JWT client, or Compute client, t
 
 For example, a JWT auth client will be created when your code is running on your local developer machine, and a Compute client will be created when the same code is running on Google Cloud Platform.
 
-The code below shows how to retrieve a default credential type, depending upon the runtime environment. The createScopedRequired must be called to determine when you need to pass in the scopes manually, and when they have been set for you automatically based on the configured runtime environment.
+The code below shows how to retrieve a default credential type, depending upon the runtime environment.
```
Comment (jinwoo) [inline README.md]: It'd be good to mention that `scopes` can be either a string array or just a string.

File: test/test.googleauth.ts (hunk 1 above)
Comment (JustinBeckwith) [reply test/test.googleauth.ts]: good call :)

File: test/test.googleauth.ts [hunk 3]
```diff
@@ -1233,6 +1233,13 @@ it('should allow passing scopes to get a client', async () => {
   assert.equal(client.scopes, scopes);
 });
 
+it('should allow passing a scope to get a client', async () => {
```
Comment (jinwoo) [inline test/test.googleauth.ts]: Better to have a different description than the test above.

File: test/test.googleauth.ts [hunk 4]
```diff
@@ -1225,6 +1226,20 @@ it('should accept credentials to get a client', async () => {
...
 });
 
+it('should allow passing scopes to get a client', async () => {
+  const scopes = ['http://examples.com/is/a/scope'];
+  const keyFilename = './test/fixtures/private.json';
+  const client = await auth.getClient({scopes, keyFilename}) as JWT;
+  assert.equal(client.scopes, scopes);
+});
+
+it('should allow passing a scope to get a client', async () => {
```
Comment (jinwoo) [inline test/test.googleauth.ts]: Put the comment in a wrong commit. Trying again.
Better to have a description than the test above.

**Pull request merged**